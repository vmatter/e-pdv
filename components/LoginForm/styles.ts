import styled from '@emotion/styled';

export const FormWrapper = styled.section`
  max-width: 480px;
  width: 100%;
  height: 100vh;
  padding: 0 24px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const InputWrapper = styled.form`
  display: flex;
  flex-direction: column;
  padding: 12px;
  gap: 12px;
`;
