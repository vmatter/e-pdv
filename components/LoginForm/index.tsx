import { Button, TextField, Typography } from '@material-ui/core';
import { useRouter } from 'next/router';
import { useState, useEffect } from 'react';
import { FormWrapper, InputWrapper } from './styles';

const LoginForm = () => {
  const router = useRouter();
  const [loggedIn, setLoggedIn] = useState(false);
  const [loading, setLoading] = useState(false);
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  useEffect(() => {
    loggedIn && router.push('/');
  }, [loggedIn]);

  const onSubmit = async (e: any) => {
    e?.preventDefault();

    setLoading(true);
    console.log(`email`, email);
    console.log(`password`, password);

    setTimeout(() => {
      setLoading(false);
      setLoggedIn(true);
    }, 2000);
  };

  return (
    <FormWrapper>
      <Typography variant="h5" component="h2">
        e-PDV Sign In
      </Typography>
      <InputWrapper>
        <TextField
          id="input-email"
          data-testid="input-email"
          variant="outlined"
          type="text"
          name="username"
          label="username"
          onChange={(e: any) => {
            setEmail(e.target.value);
          }}
        />
        <TextField
          id="input-password"
          data-testid="input-password"
          variant="outlined"
          type="password"
          name="password"
          label="password"
          onChange={(e: any) => {
            setPassword(e.target.value);
          }}
        />
        <Button
          id="button-login"
          variant="contained"
          color="primary"
          disabled={loading}
          onClick={onSubmit}
        >
          Login
        </Button>
      </InputWrapper>
    </FormWrapper>
  );
};

export default LoginForm;
