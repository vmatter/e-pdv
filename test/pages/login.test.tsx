import React from 'react';
import LoginForm from '../../components/LoginForm';
import { render, fireEvent, waitFor } from '../testUtils';
import renderer from 'react-test-renderer';

jest.mock('next/router');

describe('Login', () => {
  //useRouter.mockReturnValue({push: expectedRouterPush});

  const expectedRouterPush = jest.fn();
  const expectedLogin = jest.fn();
  const expectedEmail = 'admin';
  const expectedPassword = 'admin';

  it('should redirect on sign in', async () => {
    const { getByTestId } = render(<LoginForm />);

    fireEvent.change(getByTestId('input-user'), {
      target: { value: expectedEmail },
    });
    fireEvent.change(getByTestId('input-password'), {
      target: { value: expectedPassword },
    });
    fireEvent.click(getByTestId('button-login'));

    await waitFor(() => {
      expect(expectedLogin).toHaveBeenCalledTimes(1);
      expect(expectedLogin).toHaveBeenCalledWith('/login');

      expect(expectedRouterPush).toHaveBeenCalledTimes(1);
      expect(expectedRouterPush).toHaveBeenCalledWith('/login');
    });
  });

  it('should show toast error', async () => {
    expectedLogin.mockRejectedValue({
      message: 'Usuário ou senha não encontrados, tente novamente.',
    });

    const { getByTestId, getByText } = render(<LoginForm />);
    fireEvent.change(getByTestId('input-user'), {
      target: { value: 'foo' },
    });
    fireEvent.change(getByTestId('input-password'), {
      target: { value: expectedPassword },
    });
    fireEvent.click(getByTestId('button-login'));

    await waitFor(() => {
      expect(expectedLogin).toHaveBeenCalledTimes(1);
      expect(expectedLogin).toHaveBeenCalledWith('foo', expectedPassword);

      const errorMessage = getByText(
        'Usuário ou senha não encontrados, tente novamente.'
      );

      expect(errorMessage).toBeInTheDocument();
    });
  });

  it('should show erros for required fields', async () => {
    const { getByTestId, getAllByText } = render(<LoginForm />);

    fireEvent.click(getByTestId('button-login'));

    await waitFor(() => {
      expect(
        getAllByText('Este campo deve ser preenchido.')
      ).toBeInTheDocument();

      expect(expectedLogin).not.toHaveBeenCalled();
      expect(expectedRouterPush).not.toHaveBeenCalled();
    });
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<LoginForm />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<LoginForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
