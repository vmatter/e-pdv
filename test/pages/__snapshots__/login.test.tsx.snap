// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  overflow: hidden;
}

.emotion-2 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  border: 1px solid rgba(0, 0, 0, 0.12);
  overflow: hidden;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-5 {
  padding: 40px 25px;
}

.emotion-7 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  padding-bottom: 8px;
}

.emotion-8 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
}

.emotion-9 {
  height: 100%;
  margin: auto 25px;
}

.emotion-11 {
  height: 45px;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 40px 25px;
  gap: 12px;
}

.emotion-15 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-16 {
  color: rgba(0, 0, 0, 0.6);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  padding: 0;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: calc(100% - 24px);
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(14px, 16px) scale(1);
  -moz-transform: translate(14px, 16px) scale(1);
  -ms-transform: translate(14px, 16px) scale(1);
  transform: translate(14px, 16px) scale(1);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
  pointer-events: none;
}

.emotion-16.Mui-focused {
  color: #556cd6;
}

.emotion-16.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-16.Mui-error {
  color: #ff1744;
}

.emotion-17 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  position: relative;
  border-radius: 4px;
}

.emotion-17.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-17:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-17:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-17.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #556cd6;
  border-width: 2px;
}

.emotion-17.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #ff1744;
}

.emotion-17.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-18 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding: 16.5px 14px;
}

.emotion-18::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-18::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-18:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-18::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-18:focus {
  outline: 0;
}

.emotion-18:invalid {
  box-shadow: none;
}

.emotion-18::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-18::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-18::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-18:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-18::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-18:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-18:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-18:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-18:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-18.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-18:-webkit-auto-fill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-18:-webkit-autofill {
  border-radius: inherit;
}

.emotion-19 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-20 {
  display: block;
  width: auto;
  padding: 0;
  height: 11px;
  font-size: 0.75em;
  visibility: hidden;
  max-width: 0.01px;
  -webkit-transition: max-width 50ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: max-width 50ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-20>span {
  padding-left: 5px;
  padding-right: 5px;
  display: inline-block;
}

.emotion-27 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #556cd6;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
}

.emotion-27::-moz-focus-inner {
  border-style: none;
}

.emotion-27.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-27 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-27:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(59, 75, 149);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-27:hover {
    background-color: #556cd6;
  }
}

.emotion-27:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-27.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-27.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-28 {
  width: 100%;
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

.emotion-29 {
  padding: 0 2rem 2rem;
}

.emotion-31 {
  margin: 0;
  color: #556cd6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-31:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
    class="emotion-0 emotion-1"
  >
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root emotion-2"
    >
      <div
        class="emotion-3 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 emotion-7"
          >
            Log in
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-8"
          >
            Faça log in na plataforma e-PDV
          </p>
        </div>
        <a
          class="emotion-9 emotion-10"
        >
          <img
            class="emotion-11 emotion-12"
            src="/pdv-logo.png"
          />
        </a>
      </div>
      <form
        autocomplete="off"
        class="emotion-13 emotion-14"
        novalidate=""
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root emotion-15"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary emotion-16"
            data-shrink="false"
            for="input-email"
            id="input-email-label"
          >
            email
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl emotion-17"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input emotion-18"
              data-testid="input-email"
              id="input-email"
              name="email"
              placeholder="ex: johndoe@gmail.com"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline emotion-19"
            >
              <legend
                class="emotion-20"
              >
                <span>
                  email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root emotion-15"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary emotion-16"
            data-shrink="false"
            for="input-password"
            id="input-password-label"
          >
            senha
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl emotion-17"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input emotion-18"
              data-testid="input-password"
              id="input-password"
              name="password"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline emotion-19"
            >
              <legend
                class="emotion-20"
              >
                <span>
                  senha
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root emotion-27"
          data-testid="button-login"
          id="button-login"
          tabindex="0"
        >
          <span
            class="MuiButton-label emotion-28"
          >
            Log In
          </span>
        </button>
      </form>
      <div
        class="emotion-29 emotion-30"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-31"
          href="/account"
        >
          Criar nova conta
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
